File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;

        for (auto c : s) {
            if (c == '(' || c == '[' || c == '{') 
                stack.push(c);
            else if (c == ')') {
                if (stack.size() == 0) return false;
                if (stack.top() == '(') stack.pop();
                else return false;
            }
            else if (c == ']') {
                if (stack.size() == 0) return false;
                if (stack.top() == '[') stack.pop();
                else return false;
            }
            else if (c == '}') {
                if (stack.size() == 0) return false;
                if (stack.top() == '{') stack.pop();
                else return false;
            }
        }
        return !stack.size();
    }
};

File: Data Structures & Algorithms/minimum-stack/submission-2.py
class MinStack:

    
    
    def __init__(self):
        self.stack = []
        self.minValues = []    

    def push(self, val: int) -> None:
        self.stack.append(val)
        val = min(val, self.minValues[-1] if self.minValues else val)
        self.minValues.append(val)

    def pop(self) -> None:
        self.stack.pop()
        self.minValues.pop()

    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.minValues[-1]
        
